plugins/rest/azure.go

```diff
@@ -55,15 +58,25 @@ func (ap *azureManagedIdentitiesAuthPlugin) NewClient(c Config) (*http.Client, e
 	}
 
 	if ap.Endpoint == "" {
-		ap.Endpoint = azureIMDSEndpoint
+		identityEndpoint := os.Getenv("IDENTITY_ENDPOINT")
```

Comment:
Are there any docs or links we can add here that point to the ordering used here? I mean if `IDENTITY_ENDPOINT` is set, does that determine the flow?

Reply:
I checked the Azure SDK implementation as a reference.

https://github.com/Azure/azure-sdk-for-go/blob/910af976fc046f50b1a6ed443c0977c7c4b55ec9/sdk/azidentity/managed_identity_client.go#L141-L183